### ISSUE TITLE
Only ticked state announced when following links targeting checkboxes or radios

### ISSUE THREAD

**Issue by 36degrees** [alphagov/reported-bugs#4]
**Upstream bug:** https://bugs.webkit.org/show_bug.cgi?id=191748
**Date:** 2018-11-16
**Reported by:** Oliver Byford
**Related to:** https://github.com/alphagov/govuk-frontend/pull/1056#issuecomment-439381905

## Overview

Given the following markup:

```
  <a href="#checkboxes-1">Link to checkboxes</a><br>
  <a href="#radios-1">Link to radios</a><br>

  <br><br>

  <fieldset aria-describedby="checkboxes-description">
    <legend>Legend for checkboxes</legend>
    <span id="checkboxes-description">
      Description for checkboxes
    </span>

    <br>

    <input id="checkboxes-1" name="checkboxes" type="checkbox" value="one">
    <label for="checkboxes-1">
      One
    </label>

    <input id="checkboxes-2" name="checkboxes" type="checkbox" value="two">
    <label for="checkboxes-2">
      Two
    </label>
  </fieldset>

  <br><br>

  <fieldset aria-describedby="radios-description">
    <legend>Legend for radios</legend>
    <span id="radios-description">
      Description for radios
    </span>

    <br>

    <input id="radios-1" name="radios" type="radio" value="one">
    <label for="radios-1">
      One
    </label>
      
    <input id="radios-2" name="radios" type="radio" value="two">
    <label for="radios-2">
      Two
    </label>
  </fieldset>
```

Focus the link to each input type and activate it by double-tapping.

### Expected behaviour:

Voiceover should announce the same context that it announces when reaching the first checkbox in the fieldset by swiping in either direction through the document, including the label, field type, legend and any description:

"Legend for checkboxes, Description for checkboxes, Two, tickbox, unticked. Description for checkboxes. Double-tap to toggle setting."

(The description is repeated here, which is probably another bug...)

### Actual behaviour:

Voiceover does not include the legend as part of the announcement, and only announces the ticked state:

"Unticked"

### Expected behaviour:

Voiceover should announce the same context that it announces when reaching the first checkbox in the fieldset by swiping in either direction through the document, including the label, field type, legend and any description:

"Legend for radios, Description for radios, Two, radio button, unticked. Two of two. Description for radios."

(Again, the description is repeated here)

### Actual behaviour:

Voiceover does not include the legend as part of the announcement, and only announces the ticked state:

"Unticked"

Tested in iOS 12.1 (16B92) on an iPhone X (A1901)

This may be related to https://bugs.webkit.org/show_bug.cgi?id=191746.

**Comment by querkmachine** [comment]
Re-tested in iOS 26.4 and this is still an issue.
